File: Problems/Leetcode/3001-4000/3201-3300/3255.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;

static constexpr int MOD = 1e9 + 7;

/*
    更简洁的写法
*/

class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        int n = nums.size();
        vector<int> ans(n - k + 1, -1);
        int cnt = 0;
        for (int i = 0; i < n; ++i) {
            cnt = i == 0 || nums[i] == nums[i - 1] + 1 ? cnt + 1 : 1;
            if (cnt >= k) {
                ans[i - k + 1] = nums[i];
            }
        }
        return ans;
    }
};

int main() {
    Solution sol;

    return 0;
}

File: Problems/Leetcode/3001-4000/3201-3300/3255.py
from typing import List


class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        ans = [-1] * (n - k + 1)
        cnt = 0
        for i in range(n):
            if i == 0 or nums[i] == nums[i - 1] + 1:
                cnt += 1
            else:
                cnt = 1
            if cnt >= k:
                ans[i - k + 1] = nums[i]
        return ans
